docs/tutorials/using_pyrokinetics_on_hpc_systems.rst

```diff
@@ -0,0 +1,395 @@
+
+.. _sec-simple-hpc:
+
+Simple Usage
+------------
+
+As a simple example, let's say we want to perform a bunch of similar simulations over
+a range of proposed equilibrium geometries, such as those available
+`here <https://doi.org/10.5281/zenodo.4643844>`_.
+
+Rather than running Pyrokinetics directly on the login nodes, it is recommended to
+collect all commands into an ``sbatch`` script, and to submit this to the Slurm job
+scheduling system. Depending on your system, you may need to create this in your
+'scratch' or 'work' space -- the area on your file system where it's permitted to save
+large amounts of data, and where items may be periodically deleted to free up space.
+We'll work in the directory ``~/scratch/pyrokinetics/``, and we'll begin by copying over
+our equilibrium files and a GS2 template file:
+
+.. code-block:: bash
+
+   $ scp pyrokinetics_template.in *.geqdsk username@hpc.com:scratch/pyrokinetics/
+
+
+The following script at ``~/scratch/pyrokinetics/pyrokinetics.sh`` will be used to
+specify the job:
+
+.. code-block:: bash
+
+   #!/bin/bash
+
+   # Slurm settings
+   # --------------
+
+   #SBATCH --job-name=pyrokinetics  # Job name
+   #SBATCH --mail-user=me@email.com # Where to send mail reporting on the job
+   #SBATCH --mail-type=END,FAIL     # Mail events (NONE, BEGIN, END, FAIL, ALL)
+   #SBATCH --nodes=1                # Number of compute nodes to run on
+   #SBATCH --ntasks-per-node=16     # Number of MPI processes to spawn per node
+   #SBATCH --cpus-per-task=1        # Number of CPUs per MPI process
+   #SBATCH --mem-per-cpu=8gb        # Memory allocated to each CPU
+   #SBATCH --time=00:30:00          # Total time limit hrs:min:sec
+   #SBATCH --output=%x_%j.log       # Log file for stdout outputs
+   #SBATCH --error=%x_%j.err        # Log file for stderr outputs
+
+   # The following SBATCH settings will vary depending on your machine:
+
+   #SBATCH --partition=nodes
+   #SBATCH --account=myaccountname
+
+   # User settings
+   # -------------
+
+   # Set the location of the gs2 executable on your system
+   gs2=$HOME/gs2/bin/gs2
+
+   # Load the modules needed to run Pyrokinetics and GS2
+   # These will vary depending on your system, and how GS2 was compiled
+   module purge
+   module load Python OpenMPI netCDF-Fortran FFTW.MPI/3.3.10-gompi-2023a
+
+   # Activate the Python environment we installed Pyrokinetics to
+   source $HOME/.local/venv/pyrokinetics/bin/activate
+
+   # Generate inputs
+   # ---------------
+
+   pyro convert GS2 pyrokinetics_template.in --eq tdotp_lowq0.geqdsk --psi 0.5 \
+     -o tdotp_lowq0.geqdsk.d/tdotp_lowq0.in
+   pyro convert GS2 pyrokinetics_template.in --eq tdotp_highq0.geqdsk --psi 0.5 \
+     -o tdotp_highq0.geqdsk.d/tdotp_highq0.in
+   pyro convert GS2 pyrokinetics_template.in --eq tdotp_negtri.geqdsk --psi 0.5 \
+     -o tdotp_negtri.geqdsk.d/tdotp_negtri.in
+
+   # Perform runs
+   # ------------
+
+   srun $gs2 tdotp_lowq0.geqdsk.d/tdotp_lowq0.in
```

Comment:
Note slurm has a (sysadmin configurable) maximum number of srun commands which can appear in a single job submission. The default is something like 40,000 so unlikely to hit this, but might be worth a mention somewhere?